File: 1365-how-many-numbers-are-smaller-than-the-current-number/1365-how-many-numbers-are-smaller-than-the-current-number.java
class Solution {
    public int[] smallerNumbersThanCurrent(int[] nums) {
        
        int[] arr = new int[nums.length];
        
            for(int i=0; i<=nums.length-1; i++){
                
                int countHighest=0;
                
                int check = nums[i];
                
                for(int j=0; j<=nums.length-1; j++){
                
                if(j!=i && nums[j] < nums[i]){
                    countHighest += 1; 
                }    
                    
                }  
                
                arr[i] = countHighest;       
            }
        
        return arr;
        
    }
}

File: 1365-how-many-numbers-are-smaller-than-the-current-number/1365-how-many-numbers-are-smaller-than-the-current-number.py
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        res=[]
        
        for i in nums:
            cnt=0
            for j in nums:
                if i > j:
                    cnt+=1
            res.append(cnt)
        return res
    
 #Optimized solution using hashmap to store the index position of sorted list, which itself gives the freq of smaller numbers  
#     	temp = sorted(nums)
# 	    mapping = {}
# 	    result = []
# 	    for i in range(len(temp)):
# 	    	if temp[i] not in mapping:
# 	    		mapping[temp[i]] = i
# 	    for i in range(len(nums)):
# 	    	result.append(mapping[nums[i]])
# 	    return result